### DISCUSSION TITLE
How to upgrade Nightingale

### DISCUSSION THREAD

**Discussion by UlricQin** [ccfos/nightingale#1194] [Show and tell]
一个程序要想运行，主要是二进制、配置、数据、前端静态资源文件。

升级夜莺，一般二进制可以直接替换为新版本的二进制；配置文件有可能变化有可能没变，可以找一些文件对比工具，比如vimdiff，对本地配置文件和发布包里的最新的配置文件做对比，看看改了哪些地方，然后做相应的调整即可；数据的话显然不会变，但是数据库表结构有可能会变，参考各个版本的changelog：https://github.com/ccfos/nightingale/releases 每个版本发布如果需要修改数据库表结构，changelog里都会写清楚的，如果你当前是5.4.0版本，你就从这个版本开始逐步往更高的版本查看，5.4.1、5.4.2、5.5.0等等，挨个查看对应版本的changelog，看看有没有sql要执行，有的话就挨个执行；最后是前端静态资源文件，直接替换pub目录即可；最后重启服务即可完成。

当然，升级之前要先备份，这是好习惯。万一中间搞错了，还可以回滚。

**Comment by DerickIT** [comment]
所以是建议我们采用二进制的方式安装吗

**Reply by UlricQin** [reply]
高手，啥方式都行，小白，建议二进制方式
